src/mesh.hpp

```diff
@@ -78,13 +79,20 @@ enum datatype { READ,
  */
 struct MeshData {
   MeshData(datatype type, int numcomp, std::string name, int dataID)
-      : type(type), numcomp(numcomp), name(std::move(name)), dataID(dataID) {}
+      : type(type), numcomp(numcomp), name(std::move(name)), dataID(dataID){};
+#ifdef ASTE_NN_GRADIENT_MAPPING
+  MeshData(datatype type, int numcomp, std::string name, int dataID, int gradDimension)
+      : type(type), numcomp(numcomp), name(std::move(name)), dataID(dataID), gradDimension(gradDimension){};
+#endif
 
   datatype            type;
   int                 numcomp;
   std::string         name; // name of data
   std::vector<double> dataVector;
   int                 dataID; // preCICE dataID
+#ifdef ASTE_NN_GRADIENT_MAPPING
```

Comment:
As long as we don't involve any preCICE API calls, we can simply define such variables without the PP variable clutter. We just need to ensure at some place that we don't enter the gradient branch.